library/core/src/task/wake.rs

```diff
@@ -195,16 +236,43 @@ impl<'a> Context<'a> {
     #[must_use]
     #[inline]
     pub const fn from_waker(waker: &'a Waker) -> Self {
-        Context { waker, _marker: PhantomData, _marker2: PhantomData }
+        ContextBuilder::from_waker(waker).build()
     }
 
     /// Returns a reference to the [`Waker`] for the current task.
+    ///
+    /// Note that if the waker does not need to be sent across threads, it
+    /// is preferable to call `local_waker`, which is more portable and
+    /// potentially more efficient.
```

Comment:
I'm not sure we should say this while the API remains unstable. I'm not sure how much precedent we have, but I feel like we generally try to avoid docs talking too much about unstable area. This seems like a particularly clear case where *right now* it's quite unlikely that any of the surface area added in this PR is what you should use "by default".